azext_iot/common/certops.py

```diff
@@ -102,6 +102,51 @@ def create_self_signed_certificate(
     return result
 
 
+def isBase64(content: str) -> bool:
+    """
+    Checks if certificant content should be valid base64 string without prefix and suffix
+
+    Args:
+        content (str): certificant content without prefix and suffix.
+
+    Returns:
+        isBase64 (bool): returns where the certificant content is valid base64 value.
+    """
+    try:
+        sb_bytes = bytes(content, "ascii")
+        base64.b64decode(sb_bytes)
+    except Exception:
+        return False
+    return True
+
+
+def getCertificateFormatValidation(certificate: str) -> str:
+    """
+    Checks if the certificate format is valid
+    1. start with -----BEGIN CERTIFICATE----- (prefix)
+    2. end with -----END CERTIFICATE----- (suffix)
+    3. content should be valid base64 string without prefix and suffix
+
+    Args:
+        certificate (str): certificate string.
+
+    Returns:
+        validation string (str): returns validation string when content format is incorrect.
+    """
+    if (
+        certificate.find("-----BEGIN CERTIFICATE-----") != -1
+        and certificate.find("-----END CERTIFICATE-----") != -1
+    ):
+        certificate = certificate.replace("-----BEGIN CERTIFICATE-----", "")
+        certificate = certificate.replace("-----END CERTIFICATE-----", "")
+        if isBase64(certificate):
+            return ""
+        else:
+            return "The certificate content is not a valid base64 string value"
+    else:
+        return "The certificate should start with '-----BEGIN CERTIFICATE-----' and end with '-----END CERTIFICATE-----'"
```

Comment:
Related to my other comment -- I'm wondering if we can test when the BEGIN and END segments are missing and also if they are part of the base64 payload encoding directly. IIRC we ran into a user issue around this, which caused us to make the `except UnicodeError:` change (`certificate = base64.b64encode(certificate).decode("utf-8")`) in the open_certificates() function. 

@c-ryan-k has context on this.

Reply:
Thanks for providing the context here, I tested 5 scenarios for the service call:
1. _BEGIN CERTIFICATE + base64 content + END CERTIFICATE_: passed
2. _base64 content only_: passed
3. _EncodeBase64(BEGIN CERTIFICATE + base64 content + END CERTIFICATE)_: passed
4. _base64 content + BEGIN CERTIFICATE/END CERTIFICATE_: failed
5. _EncodeBase64(base64 content + BEGIN CERTIFICATE/END CERTIFICATE)_: passed

seems like the service call accept most of the content and shows clear explanation when there is error. Should I just remove the content validation on client side? or I can change to just validate 4th scenario

Reply:
I'm thinking to change the validator logic to below:
1. In certificate string search for "CERTIFICATE" keyword, if there is, then verify if we got matching prefix and suffix
2. if there is no "CERTIFICATE", verify if it's base64 decodable